src/NHibernate/Dialect/Function/CastFunction.cs

```diff
@@ -50,13 +50,8 @@ public SqlString Render(IList args, ISessionFactoryImplementor factory)
 				{
 					throw new QueryException("invalid NHibernate type for cast(), was:" + typeName);
 				}
+
 				sqlType = factory.Dialect.GetCastTypeName(sqlTypeCodes[0]);
-				if (sqlType == null)
-				{
-					//TODO: never reached, since GetTypeName() actually throws an exception!
```

Comment:
It doesn't make sense using the type name when the dialect doesn't support the cast. If the cast would be supported, `GetCastTypeName` should return it.